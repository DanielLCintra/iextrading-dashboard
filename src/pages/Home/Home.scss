.header {
  padding: 100px;
}

.stocksList {
  margin: 0 100px;
}
