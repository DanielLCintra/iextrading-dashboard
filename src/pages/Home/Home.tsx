import * as React from 'react';
import Header from '../../components/Header';
import StockCard from '../../components/StockCard';
import { IStock, IStocksState } from '../../ducks/stocks';
import classes from './Home.scss';

export interface IHomeProps {
  loadStocks: () => any;
  stocks: IStocksState,
}

class Home extends React.Component<any, IHomeProps> {
  public componentDidMount() {
    this.props.loadStocks();
  }

  public render() {
    const stocks = this.props.stocks.data || [];
    return (
      <div className={classes.wrapper}>
        <Header />
        <div className={classes.stocksList}>
          {
            stocks.map((stock:IStock) => (
              <StockCard
                stock={stock}
                key={stock.symbol}
              />
            ))
          }
        </div>
      </div>
    );
  }
}

export default Home;
