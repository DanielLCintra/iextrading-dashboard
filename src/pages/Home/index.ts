import { connect } from 'react-redux';
import { DUCK_NAME as STOCKS_DUCK_NAME, loadStocks } from '../../ducks/stocks';
import Home from './Home';

const mapStateToProps = (state:any) => ({
  stocks: state[STOCKS_DUCK_NAME],
});

const mapDispatchToProps = (dispatch:any) => ({
  loadStocks: () => dispatch(loadStocks()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);