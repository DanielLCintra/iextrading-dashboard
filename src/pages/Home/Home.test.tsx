import { shallow } from 'enzyme';
import * as React from 'react';
import Home, { IHomeProps } from './Home';

describe('Home Page', () => {
  const mockLoadStocks = jest.fn();
  const props:IHomeProps = {
    loadStocks: mockLoadStocks,
    stocks: {
      data: null,
      error: null,
      loading: false,
    }
  };

  const wrapper = shallow(<Home {...props} />);

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should try to load stocks', () => {
    wrapper.shallow();
    expect(mockLoadStocks).toBeCalled();
  });
});
