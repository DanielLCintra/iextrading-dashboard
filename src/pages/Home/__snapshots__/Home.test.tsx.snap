// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders without crashing 1`] = `
<div>
  <Header />
  <div />
</div>
`;
